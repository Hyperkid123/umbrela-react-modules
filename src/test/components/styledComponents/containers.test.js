import React from 'react'
import {
  QuestionListItem,
  ChartContainer,
  PaginationsListItem,
  MatrixTableRow,
  LabelContainer,
  MenuItem,
  MenuContainer,
  FlexSection,
  Flex
} from '../../../common/styledComponents/containers';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Styled containers components', () => {
  it('should render Question list item for matrix question correctly', () => {
    const tree = shallow(
      <QuestionListItem matrix>
        <div>foo</div>
      </QuestionListItem>
    )
    expect(toJson(tree)).toMatchSnapshot()
  })

  it('should render Question list item with answer error correctly', () => {
    const tree = shallow(
      <QuestionListItem error>
        <div>foo</div>
      </QuestionListItem>
    )
    expect(toJson(tree)).toMatchSnapshot()
  })

  it('should render Chart container for pie chart correctly', () => {
    const tree = shallow(
      <ChartContainer>
        <div>foo</div>
      </ChartContainer>
    )
    expect(toJson(tree)).toMatchSnapshot()
      const pieTree = shallow(
        <ChartContainer pie>
          <div>foo</div>
        </ChartContainer>
      )
      expect(toJson(pieTree)).toMatchSnapshot()
  })

  it('should render Paginations list item with correct collors based on active state', () => {
    const disbaledTree = shallow(
      <PaginationsListItem disabled>
        <div>foo</div>
      </PaginationsListItem>
    )
    expect(toJson(disbaledTree)).toMatchSnapshot()

    const activeTree = shallow(
      <PaginationsListItem active>
        <div>foo</div>
      </PaginationsListItem>
    )
    expect(toJson(activeTree)).toMatchSnapshot()

    const tree = shallow(
      <PaginationsListItem>
        <div>foo</div>
      </PaginationsListItem>
    )
    expect(toJson(tree)).toMatchSnapshot()
  })

  it('should render matrix table row with filler cells', () => {
    const tree = shallow(
      <MatrixTableRow>
        <td>foo</td>
        <td>foo</td>
        <td>foo</td>
        <td>foo</td>
        <td>foo</td>
      </MatrixTableRow>
    )
    expect(toJson(tree)).toMatchSnapshot();

    const fillerTree = shallow(
      <MatrixTableRow filler>
        <td>foo</td>
        <td>foo</td>
        <td>foo</td>
        <td>foo</td>
        <td>foo</td>
      </MatrixTableRow>
    )
    expect(toJson(fillerTree)).toMatchSnapshot();
  })

  it('it should render Label container component with left margin', () => {
    const leftTree = shallow(
      <LabelContainer alignRight>
        foo
      </LabelContainer>
    )
    expect(toJson(leftTree)).toMatchSnapshot()
  })

  it('it should render Label container component with error', () => {
    const errorTree = shallow(
      <LabelContainer error>
        foo
      </LabelContainer>
    )
    expect(toJson(errorTree)).toMatchSnapshot()
  })

  it('should render menu item active', () => {
    const activeTree = shallow(
      <MenuItem active>
        foo
      </MenuItem>
    )
    expect(toJson(activeTree)).toMatchSnapshot();
  })


  it('should render menu item dragging', () => {
    const draggingTree = shallow(
      <MenuItem dragging>
        foo
      </MenuItem>
    )
    expect(toJson(draggingTree)).toMatchSnapshot();
  })

  it('should render Menu container full width', () => {
    const tree = shallow(
      <MenuContainer>
        foo
      </MenuContainer>
    )
    expect(toJson(tree)).toMatchSnapshot();

    const fullWidthTree = shallow(
      <MenuContainer fullWidth>
        foo
      </MenuContainer>
    )
    expect(toJson(fullWidthTree)).toMatchSnapshot();
  })

  it('should render Flex section correctly based on props', () => {
    const tree = shallow(
      <FlexSection>
        foo
      </FlexSection>
    )
    expect(toJson(tree)).toMatchSnapshot();

    const minWidthTree = shallow(
      <FlexSection minWidth={33}>
        foo
      </FlexSection>
    )
    expect(toJson(minWidthTree)).toMatchSnapshot();

    const autoHeightTree = shallow(
      <FlexSection autoHeight>
        foo
      </FlexSection>
    )
    expect(toJson(autoHeightTree)).toMatchSnapshot();

    const fullWidthtree = shallow(
      <FlexSection fullWidth>
        foo
      </FlexSection>
    )
    expect(toJson(fullWidthtree)).toMatchSnapshot();
  })

  it('should render Flex component based on props', () => {
    const tree = shallow(
      <Flex>
        foo
      </Flex>
    )
    expect(toJson(tree)).toMatchSnapshot()

    const autoWidthTree = shallow(
      <Flex auto>
        foo
      </Flex>
    )
    expect(toJson(autoWidthTree)).toMatchSnapshot()

    const baselineTree = shallow(
      <Flex baseline>
        foo
      </Flex>
    )
    expect(toJson(baselineTree)).toMatchSnapshot()


    const endTree = shallow(
      <Flex end>
        foo
      </Flex>
    )
    expect(toJson(endTree)).toMatchSnapshot()
  })
});
