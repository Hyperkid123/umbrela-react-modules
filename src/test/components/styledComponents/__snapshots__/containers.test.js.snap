// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled containers components it should render Label container component with error 1`] = `
<div
  className="sc-htoDjs dwDfmY"
>
  foo
</div>
`;

exports[`Styled containers components it should render Label container component with left margin 1`] = `
<div
  className="sc-htoDjs hlazrg"
>
  foo
</div>
`;

exports[`Styled containers components should render Chart container for pie chart correctly 1`] = `
<div
  className="sc-hSdWYo kQuTFZ"
>
  <div>
    foo
  </div>
</div>
`;

exports[`Styled containers components should render Chart container for pie chart correctly 2`] = `
<div
  className="sc-hSdWYo gpELlf"
>
  <div>
    foo
  </div>
</div>
`;

exports[`Styled containers components should render Flex component based on props 1`] = `
<div
  className="sc-bdVaJa fizmbb"
>
  foo
</div>
`;

exports[`Styled containers components should render Flex component based on props 2`] = `
<div
  className="sc-bdVaJa jZPlqj"
>
  foo
</div>
`;

exports[`Styled containers components should render Flex component based on props 3`] = `
<div
  className="sc-bdVaJa bfRLQo"
>
  foo
</div>
`;

exports[`Styled containers components should render Flex component based on props 4`] = `
<div
  className="sc-bdVaJa jHiMbo"
  end={true}
>
  foo
</div>
`;

exports[`Styled containers components should render Flex section correctly based on props 1`] = `
<section
  className="sc-bxivhb gkWBUZ"
>
  foo
</section>
`;

exports[`Styled containers components should render Flex section correctly based on props 2`] = `
<section
  className="sc-bxivhb eIAtTJ"
>
  foo
</section>
`;

exports[`Styled containers components should render Flex section correctly based on props 3`] = `
<section
  className="sc-bxivhb bDsTwg"
>
  foo
</section>
`;

exports[`Styled containers components should render Flex section correctly based on props 4`] = `
<section
  className="sc-bxivhb hNhWTE"
>
  foo
</section>
`;

exports[`Styled containers components should render Menu container full width 1`] = `
<div
  className="sc-htpNat fuGjHS"
>
  foo
</div>
`;

exports[`Styled containers components should render Menu container full width 2`] = `
<div
  className="sc-htpNat cEkwiA"
>
  foo
</div>
`;

exports[`Styled containers components should render Paginations list item with correct collors based on active state 1`] = `
<li
  className="sc-iAyFgw hWRYFs"
  disabled={true}
>
  <div>
    foo
  </div>
</li>
`;

exports[`Styled containers components should render Paginations list item with correct collors based on active state 2`] = `
<li
  className="sc-iAyFgw dUoMCh"
>
  <div>
    foo
  </div>
</li>
`;

exports[`Styled containers components should render Paginations list item with correct collors based on active state 3`] = `
<li
  className="sc-iAyFgw esHeKX"
>
  <div>
    foo
  </div>
</li>
`;

exports[`Styled containers components should render Question list item for matrix question correctly 1`] = `
<div
  className="sc-brqgnP jeCayQ"
>
  <div>
    foo
  </div>
</div>
`;

exports[`Styled containers components should render Question list item with answer error correctly 1`] = `
<div
  className="sc-brqgnP cdSsxY"
>
  <div>
    foo
  </div>
</div>
`;

exports[`Styled containers components should render matrix table row with filler cells 1`] = `
<tr
  className="sc-ckVGcZ jziLBU"
>
  <td>
    foo
  </td>
  <td>
    foo
  </td>
  <td>
    foo
  </td>
  <td>
    foo
  </td>
  <td>
    foo
  </td>
</tr>
`;

exports[`Styled containers components should render matrix table row with filler cells 2`] = `
<tr
  className="sc-ckVGcZ eMEZbW"
>
  <td>
    foo
  </td>
  <td>
    foo
  </td>
  <td>
    foo
  </td>
  <td>
    foo
  </td>
  <td>
    foo
  </td>
</tr>
`;

exports[`Styled containers components should render menu item active 1`] = `
<div
  className="sc-EHOje kOFXmI"
>
  foo
</div>
`;

exports[`Styled containers components should render menu item dragging 1`] = `
<div
  className="sc-EHOje fRSdlp"
>
  foo
</div>
`;
